.github/scripts/promote_branch.sh

```diff
@@ -90,6 +101,144 @@ check_if_any_commits_in_last_week() {
+fetch_prs_graphql() {
+    local since_date="$1"
+    local page_size="${2:-${GITHUB_GRAPHQL_PAGE_SIZE:-100}}"  # Configurable via env var
+    local cursor="${3:-null}"
+
+    # Calculate cursor parameter for pagination
+    local cursor_param=""
+    if [ "$cursor" != "null" ]; then
+        cursor_param=", after: \\\"$cursor\\\""
+    fi
+
+    # GraphQL query to fetch PRs with commits
+    local query_string="query { repository(owner: \\\"$ORG\\\", name: \\\"$REPO\\\") { pullRequests(first: $page_size, states: MERGED, orderBy: {field: UPDATED_AT, direction: DESC}$cursor_param) { pageInfo { hasNextPage endCursor } nodes { number title url mergedAt mergeCommit { oid } labels(first: 20) { nodes { name color description } } commits(first: 250) { nodes { commit { oid messageHeadline } } } } } } }"
+    local query="{\"query\": \"$query_string\"}"
+
+    # Make GraphQL request
+    local response=$(curl -s -X POST \
+        -H "Authorization: bearer $token" \
+        -H "Content-Type: application/json" \
+        --data "$query" \
+        https://api.github.com/graphql)
+
+    echo "$response"
+}
+
+# Function to get all PRs since a given date with pagination
+get_all_prs_since_date() {
+    local since_date="$1"
+    local all_prs="[]"
+    local has_next_page=true
+    local cursor="null"
+
+    echo "Fetching all merged PRs since $since_date using GraphQL..." >&2
+
+    while [ "$has_next_page" = true ]; do
+        echo "Fetching page (cursor: $cursor)..." >&2
+
+        local response=$(fetch_prs_graphql "$since_date" "${GITHUB_GRAPHQL_PAGE_SIZE:-100}" "$cursor")
+
+        # Check for GraphQL errors
+        local errors=$(echo "$response" | jq -r '.errors // empty')
+        if [ -n "$errors" ]; then
+            echo "GraphQL Error: $errors" >&2
+            exit 1
+        fi
+
+        # Extract PR data
+        local page_prs=$(echo "$response" | jq -r '.data.repository.pullRequests.nodes // []')
+
+        # Check if we got valid data
+        if [ "$page_prs" = "null" ] || [ "$page_prs" = "" ]; then
+            echo "No PR data returned from GraphQL API" >&2
+            break
+        fi
+
+        # Filter PRs by date (since GraphQL doesn't support date filtering directly)
+        local filtered_prs=$(echo "$page_prs" | jq --arg since_date "$since_date" '
+            map(select(.mergedAt and (.mergedAt | split("T")[0] >= $since_date)))
+        ')
+
+        # If no PRs match our date criteria, we can stop
+        local filtered_count=$(echo "$filtered_prs" | jq 'length')
+        if [ "$filtered_count" -eq 0 ]; then
+            echo "No more PRs found since $since_date, stopping pagination." >&2
+            break
+        fi
+
+        # Merge with existing PRs (use temp files to avoid argument length limits)
+        local temp_merge_file=$(mktemp)
+        echo "$all_prs" > "$temp_merge_file"
+        echo "$filtered_prs" >> "$temp_merge_file"
+        all_prs=$(jq -s 'add' "$temp_merge_file")
```

Comment:
And now `$all_prs` is very big and on next iteration we do `echo "$all_prs"`, so we could hit arg list too long, no? I understand that since we get 100 PRs at a time, in most cases there will be just one iteration, but still... all_prs should really be stored in a file. But if we're worried it's gonna be big, then we should use files from the start, right? Starting with the response from the curl command.